src/Agent.Worker/TestResults/TestDataPublisher.cs

```diff
@@ -64,23 +69,48 @@ public async Task<bool> PublishAsync(TestRunContext runContext, List<string> tes
                 if (testDataProvider != null){
                     var testRunData = testDataProvider.GetTestRunData();
                     //publishing run level attachment
-                    publishTasks.Add(Task.Run(() => _testRunPublisher.PublishTestRunDataAsync(runContext, _projectName, testRunData, publishOptions, cancellationToken)));
+                    Task<IList<TestRun>> publishtestRunDataTask = Task.Run(() => _testRunPublisher.PublishTestRunDataAsync(runContext, _projectName, testRunData, publishOptions, cancellationToken));
+                    Task uploadBuildDataAttachmentTask = Task.Run(() => UploadBuildDataAttachment(runContext, testDataProvider.GetBuildData(), cancellationToken));
+
+                    publishTasks.Add(publishtestRunDataTask);
 
                     //publishing build level attachment
-                    publishTasks.Add(Task.Run(() => UploadBuildDataAttachment(runContext, testDataProvider.GetBuildData(), cancellationToken)));
+                    publishTasks.Add(uploadBuildDataAttachmentTask);
 
                     await Task.WhenAll(publishTasks);
-                    _calculateTestRunSummary = _featureFlagService.GetFeatureFlagState(TestResultsConstants.CalculateTestRunSummaryFeatureFlag, TestResultsConstants.TFSServiceInstanceGuid);
 
-                    var runOutcome = GetTestRunOutcome(_executionContext, testRunData, out TestRunSummary testRunSummary);
+                    IList<TestRun> publishedRuns = publishtestRunDataTask.Result;
 
+                    _calculateTestRunSummary = _featureFlagService.GetFeatureFlagState(TestResultsConstants.CalculateTestRunSummaryFeatureFlag, TestResultsConstants.TFSServiceInstanceGuid);
+
+                    var isTestRunOutcomeFailed = GetTestRunOutcome(_executionContext, testRunData, out TestRunSummary testRunSummary);
+                    
                     // Storing testrun summary in environment variable, which will be read by PublishPipelineMetadataTask and publish to evidence store.
                     if(_calculateTestRunSummary)
                     {
                         TestResultUtils.StoreTestRunSummaryInEnvVar(_executionContext, testRunSummary, _testRunner, "PublishTestResults");
                     }
 
-                    return runOutcome;
+                    // Check failed results for flaky aware
+                    // Fallback to flaky aware if there are any failures.
+                    using (var connection = WorkerUtilities.GetVssConnection(_executionContext))
```

Comment:
can you move this to a separate method. and protect with the try catch block. 
I don't want this to cause harm to the mainline execution at any cost.

Reply:
sure, I will do that